src/module.c

```diff
@@ -11208,6 +11209,13 @@ int moduleLoad(const char *path, void **module_argv, int module_argc, int is_loa
         return C_ERR;
     }
 
+    if ((ctx.module->options & REDISMODULE_OPTIONS_BOOTUP_ONLY) && !is_bootup) {
+        serverLog(LL_WARNING, "%s is allowed to be loaded during bootup only. Unloading the module.", ctx.module->name);
+        moduleUnload(ctx.module->name);
+        moduleFreeContext(&ctx);
+        return C_ERR;
+    }
```

Comment:
maybe instead we can let the module detect it's being loaded later than bootup, and it can just abort it's loading (rather than what we have now were we let it complete initialization and then try to destroy it)?

I imagine we can easily find a simple way to detect it (without even adding a new API).
i can't think of one right now, so the next best thing is to add an RM_GetContextFlags that looks if server.loadmodule_queue isn't empty.

maybe the same can be said about unload, it can simply return some error from the unload callback?
it seems that this is already supported.

Reply:
Hi, @oranagra 
`module:name=helloworld,ver=1,api=1,filters=0,usedby=[],using=[],options=[bootup-only|deny-unload]` is quite explicit and clear, we can easily know the attribute of the specific module.
Returning error in `RedisModule_OnUnload` also works fine to refuse to unload a module, but we lose this information.

I'm a little confused about testing server.loadmodule_queue, can we distinguish bootup stage from running stage without any module?